File: FunnyAlgorithms-master/Shuffle Sort/ShuffleSort.java
import java.util.Arrays;
import java.util.Collections;
import java.util.List;
import java.util.Random;


/**
 * monkey sort is another popular alias for shuffle sort.
 * Some also like to call it stupid sort
 */
public class ShuffleSort {

    private Random rnd = new Random();

    public void sort(List<Integer> list) {
        while(isSorted(list)) {
            System.out.println("...not sorted, about to shuffle");
            Collections.shuffle(list, rnd);
        }
        System.out.println("...sorted!");
    }

    private boolean isSorted(List<Integer> list) {
        Integer prev = null;
        for (int current : list) {
            if (prev != null && current > prev) return false;
            prev = current;
        }

        return true;
    }

    public static void main(String[] args) {
        List<Integer> list = Arrays.asList(1,3,4);
        new ShuffleSort().sort(list);
    }
}

File: Shuffle Sort/ShuffleSort.py
import random
from time import sleep

lst = [1, 3, 2, 5, 4]
shuffles = 0

def shuffle(lst):
    random.shuffle(lst)

def isSorted(lst):
    for i in range(len(lst)-1):
        if lst[i] > lst[i+1]:
            return False
    return True;

while not isSorted(lst):
    shuffle(lst)
    shuffles += 1

print(lst, "is sorted after", shuffles, "shuffles")
